File: python/basic.py
from maazdb import MaazDB

# 1. Initialize the client
db = MaazDB()

try:
    # 2. Connect securely
    db.connect(host="127.0.0.1", port=8888, user="admin", password="admin")
    print("✓ Connected to MaazDB")

    # 3. Execute SQL
    db.query("CREATE DATABASE analytics;")
    db.query("USE analytics;")
    db.query("CREATE TABLE logs (id SERIAL PRIMARY KEY, message TEXT);")
    
    # 4. Insert and Fetch
    db.query("INSERT INTO logs (message) VALUES ('System started');")
    results = db.query("SELECT * FROM logs;")
    print(f"Results:\n{results}")

except Exception as e:
    print(f"An error occurred: {e}")

finally:
    # 5. Always close the connection
    db.close()

File: JavaScript/basic.js
const MaazDB = require('maazdb-js');

async function main() {
    // 1. Initialize the client
    const db = new MaazDB();

    try {
        // 2. Connect securely (TLS is handled automatically)
        await db.connect("127.0.0.1", 8888, "admin", "admin");
        console.log("✓ Connected to MaazDB");

        // 3. Run SQL commands
        await db.query("CREATE DATABASE web_app;");
        await db.query("USE web_app;");
        
        // 4. Insert Data
        await db.query("CREATE TABLE users (id SERIAL PRIMARY KEY, username TEXT);");
        await db.query("INSERT INTO users (username) VALUES ('maaz_dev');");

        // 5. Fetch Results
        const results = await db.query("SELECT * FROM users;");
        console.log("Results:", results);

    } catch (error) {
        console.error("Database Error:", error.message);
    } finally {
        // 6. Close connection
        db.close();
    }
}

main();